src/riak_ensemble_peer.erl

```diff
@@ -1850,12 +1863,17 @@ handle_event({watch_leader_status, Pid}, StateName, State) when node(Pid) =/= no
     {next_state, StateName, State};
 handle_event({watch_leader_status, Pid}, StateName, State = #state{watchers = Watchers}) ->
     _ = notify_leader_status(Pid, StateName, State),
-    %% Might as well take this opportunity to prune any dead pids that are in the list
-    NewWatcherList = [P || P <- [Pid | Watchers], is_process_alive(P)],
-    {next_state, StateName, State#state{watchers = NewWatcherList}};
+    MRef = erlang:monitor(process, Pid),
+    {next_state, StateName, State#state{watchers = [{Pid, MRef} | Watchers]}};
 handle_event({stop_watching, Pid}, StateName, State = #state{watchers = Watchers}) ->
-    NewWatcherList = lists:delete(Pid, Watchers),
-    {next_state, StateName, State#state{watchers = NewWatcherList}};
+    case lists:keytake(Pid, 1, Watchers) of
+        false ->
+            lager:warning("Tried to stop watching for pid ~p, but did not find it in watcher list"),
```

Comment:
I was on the fence about this as well. My thinking was that a `stop_watching` for a pid that's not already watching indicates a bug in the system, so we want to make sure that such a scenario is noticed. (I even considered doing a pattern match that assumes we find the tuple, so that it crashes if it can't find it, but thought that would be a bit drastic since such a scenario doesn't necessarily indicate any crashworthy problems in the ensemble peer itself.)

From an operational standpoint, you're right, your typical sysadmin might not care about this. I was logging it as a warning from a more selfish perspective of wanting someone to notice it and say "hey Basho folk, I'm getting this message, should I worry?"

I can certainly appreciate the argument for making it a debug statement though. Thoughts?